lib/fhir/core.js

```diff
@@ -461,7 +461,12 @@ module.exports = (mongo, modules) => {
                   }
 
                   const location = `/fhir/${resourceType}/${id}/_history/${resource.meta.versionId}`
-                  callback(null, { httpStatus: 200, location: location, id: id })
+                  callback(null, {
+                    httpStatus: 200,
+                    location: location,
+                    etag: resource.meta.versionId,
+                    id: id
+                  })
```

Comment:
Don't we need to return the etag on reads and searches when we are returning resources? I'm not sure how it would be useful on an updated resource?

Reply:
We can and the spec says we should, however, the problem we are trying to solve here is managing resource contention: https://www.hl7.org/fhir/http.html#concurrency which primarily affects update requests. Unfortunately, it is not within the scope to add this support across all of Hearth.

Reply:
Ok, but to prevent update contention you need to supply the resource etag along with the update in the `If-Match` header, you won't be able to do that unless you get the etag back from a read or search. I don't recall us having support for the `If-Match` header, you may have to add some basic support on the update handling for that.

Reply:
The versionId comes back in the meta field when you read or search for a resource so the etag isn't necessary on those interactions for our use case. We do need to implement If-Match on updates though.

Reply:
Yes, you right about the If-Match header, I'll add something for that now.

Reply:
Fair enough, although he correct way to implement this would be to use the header.

The `If-Match` support is really the crux of this change.

Reply:
I see that for searches the ETag shouldn't be returned and the versionId directly from the resource should be used:

> If provided, the value of the ETag SHALL match the value of the version id for the resource. Servers are allowed to generate the version id in whatever fashion that they wish, so long as they are valid according to the id data type, and are unique within the address space of all versions of the same resource. When resources are returned as part of a bundle, there is no ETag, and the versionId of the resource is used directly.

Reply:
That makes sense, otherwise you could only return one.

Reply:
Should we leave the If-match header as optional or enforce it? I'm thinking optional because of existing systems which might not have implemented it yet...such as SSSA

Reply:
I think optional is best for now.

Reply:
Support for If-Match added in this [commit](https://github.com/jembi/hearth/pull/148/commits/be8a38a5ff3fd7dc7e05da12cc08241e80d84d7c)